src/noise.cpp

```diff
@@ -503,23 +503,32 @@ void Noise::setOctaves(int octaves)
 
 void Noise::resizeNoiseBuf(bool is3d)
 {
-	//maximum possible spread value factor
+	// Maximum possible spread value factor
 	float ofactor = (np.lacunarity > 1.0) ?
 		pow(np.lacunarity, np.octaves - 1) :
 		np.lacunarity;
 
-	// noise lattice point count
+	// Noise lattice point count
 	// (int)(sz * spread * ofactor) is # of lattice points crossed due to length
 	float num_noise_points_x = sx * ofactor / np.spread.X;
 	float num_noise_points_y = sy * ofactor / np.spread.Y;
 	float num_noise_points_z = sz * ofactor / np.spread.Z;
 
-	// protect against obviously invalid parameters
+	// Protect against obviously invalid parameters
 	if (num_noise_points_x > 1000000000.f ||
-		num_noise_points_y > 1000000000.f ||
-		num_noise_points_z > 1000000000.f)
+			num_noise_points_y > 1000000000.f ||
+			num_noise_points_z > 1000000000.f)
 		throw InvalidNoiseParamsException();
 
+	// Protect against an octave having a spread < 1, causing unusable noise values
+	if (np.spread.X / ofactor < 1.0f ||
+			np.spread.Y / ofactor < 1.0f ||
+			np.spread.Z / ofactor < 1.0f) {
+		errorstream << "A noise parameter has too many octaves, "
+			"causing unusable noise values" << std::endl;
```

Comment:
shouldn't this go into the thrown exception as a message?

Reply:
Good point, but i tested only throwing the `InvalidNoiseParamsException`, and it did not result in any noticeable error message, even in verbose debug. So i assume that `InvalidNoiseParamsException` only reports to a debug build or a debugger.
It seems that an error message is needed too to make this noticeable.
https://github.com/minetest/minetest/blob/b298b0339c79db7f5b3873e73ff9ea0130f05a8a/src/exceptions.h#L110

Reply:
Sorry, i may have misunderstood. I agree that message should be added to the exception, i will update this soon. However it seems the errorstream message is also required.

Reply:
Done.